python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -2460,6 +2472,132 @@ def _free_concurrency_slots(self, run_id: str, step_key: Optional[str] = None) -
             # return the concurrency keys for the freed slots
             return [cast(str, row[1]) for row in rows]
 
+    def store_asset_check_event(self, event: EventLogEntry, event_id: Optional[int]) -> None:
+        check.inst_param(event, "event", EventLogEntry)
+        check.opt_int_param(event_id, "event_id")
+
+        if event.dagster_event_type == DagsterEventType.ASSET_CHECK_EVALUATION_PLANNED:
+            self._store_asset_check_evaluation_planned(event, event_id)
+        if event.dagster_event_type == DagsterEventType.ASSET_CHECK_EVALUATION:
+            self._update_asset_check_evaluation(event, event_id)
+
+    def _store_asset_check_evaluation_planned(
+        self, event: EventLogEntry, event_id: Optional[int]
+    ) -> None:
+        planned = cast(
+            AssetCheckEvaluationPlanned, check.not_none(event.dagster_event).event_specific_data
+        )
+        with self.index_connection() as conn:
+            conn.execute(
+                AssetCheckExecutionsTable.insert().values(
+                    asset_key=planned.asset_key.to_string(),
+                    check_name=planned.check_name,
+                    run_id=event.run_id,
+                    execution_status=AssetCheckExecutionStatus.PLANNED.value,
+                )
+            )
+
+    def _update_asset_check_evaluation(self, event: EventLogEntry, event_id: Optional[int]) -> None:
+        evaluation = cast(
+            AssetCheckEvaluation, check.not_none(event.dagster_event).event_specific_data
+        )
+        with self.index_connection() as conn:
+            conn.execute(
+                AssetCheckExecutionsTable.update()
```

Comment:
should we just rely on the fact the application logic to enforce that all checks will have a stored planned event?

With postgres/mysql, we can do something like on duplicate key update

Reply:
You could see the resulting rowcount to see if we should log an error or raise.

Reply:
checking rowcount is a good idea. If we see the error come up for whatever reason, I think we could relax the constraint and do an insert/update here